commands/settings.ts

```diff
@@ -0,0 +1,39 @@
+import { AbstractCommand, GlobalState } from './abstractCommand'
+
+export default class Settings extends AbstractCommand {
+  name() { return 'settings' }
+  help() { return 'list your current settings' }
+
+  execute(query: string, settings: GlobalState): string {
+    let filtered: any = {}
+
+    if (query) {
+      for (let c of Object.keys(settings)){
+        if (c.startsWith(query)){
+          filtered[c] = settings[c] 
+        }
+      }
+    } else {
+      filtered = settings
+    }
+
+    let maxLength = Math.max(...Object.keys(filtered).map(x => x.length))
```

Comment:
Something something functional programming 😄

Reply:
I told you - map and reduce - the only good parts :P